File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxjump = 0;
        for(int i=0;i<nums.length;i++)
        {
            if(maxjump>=i)
            {
                maxjump = Math.max(maxjump , i+nums[i]);
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 55-jump-game/jump-game.py
class Solution:
    
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n ==1:
            return True
        dp = [0] * n
        for i in range(n):
            dp[i] = min(n-1, i + nums[i])

        print(dp)
        i = 0
        farthest = 0
        while i<=farthest:
            farthest = max(farthest, dp[i])
            if farthest >= n-1:
                return True
            i+=1
    

        return False